nautobot/extras/models/roles.py

```diff
@@ -0,0 +1,61 @@
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+from nautobot.core.fields import ForeignKeyLimitedByContentTypes
+from nautobot.core.models.name_color_content_types import NameColorContentTypesModel
+from nautobot.extras.utils import RoleModelsQuery, extras_features
+
+
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "webhooks",
+)
+class Role(NameColorContentTypesModel):
+    content_types = models.ManyToManyField(
+        to=ContentType,
+        related_name="roles",
+        verbose_name="Content type(s)",
+        limit_choices_to=RoleModelsQuery(),
+        help_text="The content type(s) to which this role applies.",
+    )
+    weight = models.PositiveSmallIntegerField(null=True, blank=True)
```

Comment:
`ipam.Role` had `ordering` of `(weight, name)` rather than just `name`. Wonder whether it'd be desirable to do the same for the new Role model?